src/containers/blocks.jsx

```diff
@@ -617,7 +621,36 @@ class Blocks extends React.Component {
         );
         defineBlocks(categoryInfo.menus);
         defineBlocks(categoryInfo.blocks);
-
+        let colourPrimary = categoryInfo.color1;
+        let colourSecondary = categoryInfo.color2;
+        let colourTertiary = categoryInfo.color3;
+        let colourQuaternary = categoryInfo.color3;
+        if (this.props.theme !== DEFAULT_THEME) {
```

Comment:
You create the theme in this file (above) so can you make all of these changes in the same place where you create, instead of creating it in one place and modifying it in another?

Reply:
No, because these colors are defined by the extension being added that invoked this handler. Since extensions can be added dynamically we don't know what styles may be needed/what colors they specify when we initially inject the workspace.